docs/how_to/env_var.md

```diff
@@ -3,36 +3,52 @@ Environment Variables
+```
+
 ## Set the Number of Threads
 
-* MXNET_GPU_WORKER_NTHREADS (default=2)
-  - The maximum number of threads that do the computation job on each GPU.
-* MXNET_GPU_COPY_NTHREADS (default=1)
-  - The maximum number of threads that do the memory copy job on each GPU.
-* MXNET_CPU_WORKER_NTHREADS (default=1)
-  - The maximum number of threads that do the CPU computation job.
-* MXNET_CPU_PRIORITY_NTHREADS (default=4)
- - The number of threads given to prioritized CPU jobs.
-* MXNET_CPU_NNPACK_NTHREADS (default=4)
- - The number of threads used for NNPACK.
+* MXNET_GPU_WORKER_NTHREADS
+  - Values: Int ```(default=2)```
+  - The maximum number of streams to use on each GPU. This parameter is used to parallelize the computation within a single GPU card.
+* MXNET_GPU_COPY_NTHREADS
+  - Values: Int ```(default=1)```
+  - The maximum number of concurrent threads that do the memory copy job on each GPU.
+* MXNET_CPU_WORKER_NTHREADS
+  - Values: Int ```(default=1)```
+  - The maximum number of scheduling threads on CPU. It specifies how many operators can be run in parallel.
+* MXNET_CPU_PRIORITY_NTHREADS
+  - Values: Int ```(default=4)```
+  - The number of threads given to prioritized CPU jobs.
+* MXNET_CPU_NNPACK_NTHREADS
+  - Values: Int ```(default=4)```
+  - The number of threads used for NNPACK.
 
 ## Memory Options
 
-* MXNET_EXEC_ENABLE_INPLACE (default=true)
+* MXNET_EXEC_ENABLE_INPLACE
+  - Values: true or false ```(default=true)```
   - Whether to enable in-place optimization in symbolic execution.
-* NNVM_EXEC_MATCH_RANGE (default=16)
+* NNVM_EXEC_MATCH_RANGE
+  - Values: Int ```(default=16)```
   - The rough matching scale in the symbolic execution memory allocator.
   - Set this to 0 if you don't want to enable memory sharing between graph nodes(for debugging purposes).
-* MXNET_EXEC_NUM_TEMP (default=1)
+* MXNET_EXEC_NUM_TEMP
+  - Values: Int ```(default=1)```
   - The maximum number of temp workspaces to allocate to each device.
   - Setting this to a small number can save GPU memory. It will also likely decrease the level of parallelism, which is usually acceptable.
-* MXNET_GPU_MEM_POOL_RESERVE (default=5)
+  - This is used to get number of matching colors and in turn how much parallelism one can get in each GPU. Color based match usually costs more memory but also enables more parallelism.
+* MXNET_GPU_MEM_POOL_RESERVE
+  - Values: Int ```(default=5)```
   - The percentage of GPU memory to reserve for things other than the GPU array, such as kernel launch or cudnn handle space.
   - If you see a strange out-of-memory error from the kernel launch, after multiple iterations, try setting this to a larger value.  
 
 ## Engine Type
 
-* MXNET_ENGINE_TYPE (default=ThreadedEnginePerDevice)
+* MXNET_ENGINE_TYPE
+  - Values: String ```(default=ThreadedEnginePerDevice)```
   - The type of underlying execution engine of MXNet.
   - Choices:
     - NaiveEngine: A very simple engine that uses the master thread to do computation.
```

Comment:
Please explain NaiveEngine a little more -- the master thread to do computation Vs what?